qemu/tests/rng_host_guest_read.py

```diff
@@ -0,0 +1,72 @@
+import re
+import logging
+
+from virttest import error_context
+from virttest import utils_test
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu virtio-rng device test:
+    1) boot guest with virtio-rng device
+    2) host read random numbers in the background
+    3) guest read random data at the same time during step2
+    4) clean host read process after test
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def host_read_start(host_read_cmd):
+        """
+        Read random numbers on the host side
+        :param host_read_cmd: host read random numbers command line
+        :return: reture host_read_process
+        """
+        host_read_process = process.SubProcess(host_read_cmd, shell=True)
+        host_read_process.start()
+        return host_read_process
+
+    def is_host_read_running(target_process):
+        """
+        Check host reading test is alive.
+
+        :param target_process: targeted host reading process
+        :return: return True if host is reading random numbers;
+                 else return False
+        """
+        error_context.context("Check whether host is reading is alive",
+                              logging.info)
+        host_read_check_cmd = params.get("host_read_check_cmd")
+        output = process.system_output(host_read_check_cmd, shell=True)
+        result = re.findall(target_process, output, re.M | re.I)
+        return bool(result)
+
+    def host_read_stop(host_read_process):
```

Comment:
I think this function is a little bit overkill.
I know this probably will not happen, but what if the process were stopped before executing  `host_read_process.stop()`

Reply:
Modified.
Add a judgement condition to check if the host_read_process is still alive.
If yes,call stop() function
Thanks